feat(Toolbar): Add row wrap prop to ToolbarGroup and ToolbarItem

mfrances17 opened the pull request:
<!-- What changes are being made? Please link the issue being addressed. -->
**What**: Closes #11489

patternfly-build commented:
Preview: https://patternfly-react-pr-11559.surge.sh

mcoker commented on packages/react-core/src/components/Toolbar/ToolbarGroup.tsx:
```diff
@@ -157,6 +157,15 @@ export interface ToolbarGroupProps extends Omit<React.HTMLProps<HTMLDivElement>,
   };
+  /** Value to set for row wrapping at various breakpoints */
+  rowWrap?: {
+    default?: 'wrap' | 'wrapReverse' | 'nowrap';
+    sm?: 'wrap' | 'wrapReverse' | 'nowrap';
+    md?: 'wrap' | 'wrapReverse' | 'nowrap';
+    lg?: 'wrap' | 'wrapReverse' | 'nowrap';
+    xl?: 'wrap' | 'wrapReverse' | 'nowrap';
+    '2xl'?: 'wrap' | 'wrapReverse' | 'nowrap';
```
```suggestion
    default?: 'wrap' | 'nowrap';
    sm?: 'wrap' | 'nowrap';
    md?: 'wrap' | 'nowrap';
    lg?: 'wrap' | 'nowrap';
    xl?: 'wrap' | 'nowrap';
    '2xl'?: 'wrap' | 'nowrap';
```

mfrances17 replied:
removed, ty!

mcoker commented on packages/react-core/src/components/Toolbar/ToolbarItem.tsx:
```diff
@@ -152,6 +152,15 @@ export interface ToolbarItemProps extends React.HTMLProps<HTMLDivElement> {
   };
+  /** Value to set for row wrapping at various breakpoints */
+  rowWrap?: {
+    default?: 'wrap' | 'wrapReverse' | 'nowrap';
+    sm?: 'wrap' | 'wrapReverse' | 'nowrap';
+    md?: 'wrap' | 'wrapReverse' | 'nowrap';
+    lg?: 'wrap' | 'wrapReverse' | 'nowrap';
+    xl?: 'wrap' | 'wrapReverse' | 'nowrap';
+    '2xl'?: 'wrap' | 'wrapReverse' | 'nowrap';
```
```suggestion
    default?: 'wrap' | 'nowrap';
    sm?: 'wrap' | 'nowrap';
    md?: 'wrap' | 'nowrap';
    lg?: 'wrap' | 'nowrap';
    xl?: 'wrap' | 'nowrap';
    '2xl'?: 'wrap' | 'nowrap';
```

mfrances17 replied:
thanks, removed

mcoker reviewed (changes requested):
Just a couple of things:
* we don't support `.pf-m-wrap-reverse`
* `.pf-m-wrap` and `.pf-m-nowrap` applies to `.pf-v6-c-toolbar__content-section`, too - https://github.com/patternfly/patternfly/blob/93efaf133c72a9e1944660bd404ad8b38f850d93/src/patternfly/components/Toolbar/toolbar.scss#L360-L369

thatblindgeye reviewed (commented):
In addition to adding the prop to ToolbarContent (for the ContentSection class), could we add a simple example for that? Maybe we could update the section from "Examples with toolbar spacers" to "Examples with spacers and wrapping", then add a "Toolbar content wrapping" example to show the nowrap modifier in use (the example description could mention that by default the content section wraps).

mfrances17 commented:
thanks @mcoker & @thatblindgeye, i've added a wrap prop to `ToolbarContent`, added an example showing the behavior with `nowrap`, and changed the examples section as suggested. Added tests for ToolbarContent wrap as well, in addition to the tests I added on my previous commit for `ToolbarGroup` and `ToolbarItem` wrapping.

mcoker reviewed (approved):
LPTM!! ✅

tlabaj reviewed (changes requested):
Looks good.  Just one ask.

patternfly-build commented:
Your changes have been released in: 
 - @patternfly/react-code-editor@6.2.0-prerelease.35
 - @patternfly/react-core@6.2.0-prerelease.33
 - @patternfly/react-docs@7.2.0-prerelease.52
 - @patternfly/react-drag-drop@6.2.0-prerelease.35
 - demo-app-ts@6.0.0-prerelease.84
 - @patternfly/react-table@6.2.0-prerelease.34
 - @patternfly/react-templates@6.2.0-prerelease.34

Thanks for your contribution! :tada:
